draft-ietf-quic-http.md

```diff
@@ -182,6 +189,22 @@ itself as "hq-09-rickroll". Note that any label MUST conform to the "token"
 syntax defined in Section 3.2.6 of [RFC7230]. Experimenters are encouraged to
 coordinate their experiments on the quic@ietf.org mailing list.
 
+## Connection Reuse
+
+Once a connection exists to a server endpoint, this connection MAY be reused for
+requests with multiple different URI authority components.  The client MAY send
+any requests for which the client considers the server authoritative.
+
+This typically means that the client has received an Alt-Svc record from the
+request's origin in question which nominates the server endpoint as a valid HTTP
```

Comment:
This is potentially misleading.  Alt-Svc doesn't form the basis for authority.  I understand the intent here is to say that you don't get here without Alt-Svc, but the "This" in this sentence is immediately after the bit about authority.

Reply:
"This" does in fact refer to considering the server authoritative, so you're reading it correctly.  RFC 7838 does say that Alt-Svc provides authority.

> Alternative services are considered authoritative for an origin's resources....

Reply:
I think that we're talking past each other.  The alternative is not authoritative because of Alt-Svc, that authority derives from the certificate.  This text implies that it derives from Alt-Svc, which is dangerously misleading.

Reply:
Perhaps.  I think this is an area that's in flux right now -- example.com:443 is authoritative for https://example.com/, but the certificate validates that you're actually talking to example.com:443.  Under any previous RFC, a server isn't authoritative simply because it has the certificate.

Alt-Svc allows an authoritative server to say that otherguy.org:8080 is co-authoritative with it.  You still need to validate the cert to verify that you've successfully reached the authoritative alternative, but it *is* the Alt-Svc delegation that makes it authoritative.  The certificate simply proves identity when connecting to that authoritative endpoint.

ORIGIN makes the stronger statement that a server can be authoritative *simply* because it has the certificate, if the client chooses to play by those new rules.  That's a new assertion in RFC [TBD], but not something contained in any existing RFC that I'm aware of.

I'll add some text saying that, as required by RFC 7838, the client MUST ensure that the certificate is valid for the origin being requested.